.github/workflows/check-pull-request.yml

```diff
@@ -37,6 +37,24 @@ jobs:
         with:
           fetch-depth: 1
 
+      - name: Set up Java
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'adopt'
+          java-version: '17'
+          cache: 'gradle'
+
       - name: Run Tests using Gradle Wrapper
-        run: ./gradlew test --no-daemon
+        run: |
+          ./gradlew --no-daemon test jacocoTestReport -x assemble
+
+      - name: Publish coverage reports
+        uses: codecov/codecov-action@v4
+        with:
+          fail_ci_if_error: true 
+          files: ./bridge/build/reports/jacoco/test/jacocoTestReport.xml
```

Comment:
I see the extension let us to define if we want a file or list of files for the reports. Can you explain me why do you choose files instead of file?

Reply:
It's because codecov requires a specific file in order to generate the report, which is this .xml. And the CI it's explicitly generating this during the  `jacocoTestReport` task

Reply:
Yeah, I see your point but you didn't catch what I'm saying. When you pick files option, you can generate more than one .xml file, but in this case we are only generating one so I'd be better to use file option.

Reply:
oh I get it now

Reply:
Nice